c/iotdm_client/lwm2m_objects/device_object.h

```diff
@@ -17,6 +17,13 @@
 
 #include "iothub_client.h"
 
+#ifdef __MACH__
```

Comment:
I'm curious why you needed to add these headers here.  There's nothing platform specific in them and I would expect them to be necessary for all of the _object.h files, not just device_object.h